api/sip/client.py

```diff
@@ -779,13 +785,13 @@ def read_message(self, max_size=1024*1024):
         A SIP2 message ends with a \\r character.
         """
         done = False
-        data = ""
+        data = b""
```

Comment:
This method contains the other essential Py3 fixes. This method wasn't previously tested at all so I added coverage.